### PR TITLE
go version 1.21, libp2p and other updates, linter update

### PR DESCRIPTION
**Pull request by dimalinux**
Updates to go 1.21, which libp2p wasn't yet ready for in the last maintenance update. As part of the lib2p2 updates, we no longer use a persistent peer store. Linter is updated to v1.54.2.

### CODE REVIEW

File: protocol/swap/swap_manager.go
Comment (dimalinux) [inline protocol/swap/swap_manager.go:193]: This change was only to make the linter happy. The original code looks safe to me.

File: rpc/server.go
Comment (dimalinux) [inline rpc/server.go:162]: This was just narrowing the scope NewPrometheusRegistry so bootnodes never create it and then not use it.

At the time I was reviewing this code, I was trying to figure out why the integration test bootnode was listening on a random 0.0.0.0 port. It turned out to be a combination of the latest libp2p code and the fact that we didn't configure the bootnode in the integration tests to not connect to the default bootnodes.

File: scripts/run-integration-tests.sh
Comment (dimalinux) [inline scripts/run-integration-tests.sh:46]: This was the needed change that was holding up the pull request. The integration test bootnode was communicating with our production bootnodes. (Remember that bootnodes no longer have the concept of an environment.) We are setting the libp2p IP below to localhost which wouldn't be able to connect to the public bootnodes, and I don't believe the connections were happening with the older libp2p versions.

File: go.mod
Comment (dimalinux) [inline go.mod:15]: There is a v1.13.5 of go-ethereum, but it is adversely affecting unit test `TestSwapCreator_Refund_afterT2`.  This PR is for libp2p updates, so I didn't look into why.

File: go.mod
Comment (dimalinux) [inline go.mod:11]: Will update to a tagged release before merging.

Comment (noot) [reply go.mod:11]: approved the go-p2p-net PR!

File: rpc/server.go
Comment (noot) [reply rpc/server.go:162]: bootnodes should probably have metrics as well - however yeah in the previous logic, that was not the case. maybe we can re-add in another PR

**Review by noot — APPROVED** [review]

**Pull request merged**
